### ISSUE TITLE
Differing auc scores between xgboost and sklearn auc metric. What could be the reason? 

### ISSUE THREAD

**Issue by nareshshah139** [dmlc/xgboost#1878]
I'm working on some private datasets and having difficulties in figuring out how to best debug this:

The auc when I run xgboost seems to show that it is 0.9+

The auc when I use sklearn.metrics.auc seem to show auc to be 0.76. 

Given that the dataset I am working on is private and cannot be shared, how do I make this reproducible? And how do I debug this?

**Comment by nareshshah139** [comment]
Some code and debug attempts:

from xgboost import XGBClassifier
from imblearn.over_sampling import SMOTE

new_list = X_train.columns

sm = SMOTE(kind='borderline1')
X_resampled2,y_resampled2 = sm.fit_sample(X_train,y_train)

X_resampled2 = pd.DataFrame(X_resampled2)
X_resampled2.columns = new_list

learning_rate = 0.1
n_estimators = 800
max_depth = 12
subsample = 0.9
colsample_bytree = 0.85
min_child_weight = 1  


xgb_model = XGBClassifier(seed=0, learning_rate=learning_rate, n_estimators=n_estimators,
                    min_child_weight=min_child_weight, max_depth=max_depth,
                    colsample_bytree=colsample_bytree, subsample=subsample, nthread = -1)

eval_metrics = ['auc']
eval_sets = [(X_resampled2 ,y_resampled2),(X_test, y_test)]



xgb = xgb_model.fit(X_resampled2, y_resampled2, eval_metric=eval_metrics, eval_set=eval_sets,early_stopping_rounds = 25 , verbose=True)

[0]	validation_0-auc:0.988404	validation_1-auc:0.896171
Multiple eval metrics have been passed: 'validation_1-auc' will be used for early stopping.

Will train until validation_1-auc hasn't improved in 25 rounds.
[1]	validation_0-auc:0.995115	validation_1-auc:0.850152
[2]	validation_0-auc:0.996226	validation_1-auc:0.847122
[3]	validation_0-auc:0.996947	validation_1-auc:0.839637
[4]	validation_0-auc:0.998572	validation_1-auc:0.876181
[5]	validation_0-auc:0.999801	validation_1-auc:0.875678
[6]	validation_0-auc:0.999839	validation_1-auc:0.889942
[7]	validation_0-auc:0.999874	validation_1-auc:0.886204
[8]	validation_0-auc:0.99994	validation_1-auc:0.903662
[9]	validation_0-auc:0.999958	validation_1-auc:0.908562
[10]	validation_0-auc:0.999972	validation_1-auc:0.911967
[11]	validation_0-auc:0.999973	validation_1-auc:0.915772
[12]	validation_0-auc:0.999978	validation_1-auc:0.914014
[13]	validation_0-auc:0.999981	validation_1-auc:0.918361
[14]	validation_0-auc:0.999982	validation_1-auc:0.918976
[15]	validation_0-auc:0.999984	validation_1-auc:0.918545
[16]	validation_0-auc:0.999987	validation_1-auc:0.918766
[17]	validation_0-auc:0.999987	validation_1-auc:0.922571
[18]	validation_0-auc:0.999994	validation_1-auc:0.922554
[19]	validation_0-auc:0.999996	validation_1-auc:0.922834
[20]	validation_0-auc:0.999997	validation_1-auc:0.919817
[21]	validation_0-auc:0.999997	validation_1-auc:0.919696
[22]	validation_0-auc:0.999998	validation_1-auc:0.914707
[23]	validation_0-auc:0.999998	validation_1-auc:0.915064
[24]	validation_0-auc:0.999998	validation_1-auc:0.91448
[25]	validation_0-auc:0.999999	validation_1-auc:0.913922
[26]	validation_0-auc:0.999999	validation_1-auc:0.914427
[27]	validation_0-auc:0.999999	validation_1-auc:0.913732
[28]	validation_0-auc:0.999999	validation_1-auc:0.913695
[29]	validation_0-auc:0.999999	validation_1-auc:0.918385
[30]	validation_0-auc:0.999999	validation_1-auc:0.918494
[31]	validation_0-auc:0.999999	validation_1-auc:0.918244
[32]	validation_0-auc:0.999999	validation_1-auc:0.918087
[33]	validation_0-auc:0.999999	validation_1-auc:0.918012
[34]	validation_0-auc:0.999999	validation_1-auc:0.917682
[35]	validation_0-auc:0.999999	validation_1-auc:0.917578
[36]	validation_0-auc:1	validation_1-auc:0.908717
[37]	validation_0-auc:1	validation_1-auc:0.907729
[38]	validation_0-auc:1	validation_1-auc:0.917301
[39]	validation_0-auc:1	validation_1-auc:0.91691
[40]	validation_0-auc:1	validation_1-auc:0.915089
[41]	validation_0-auc:1	validation_1-auc:0.91135
[42]	validation_0-auc:1	validation_1-auc:0.909446
[43]	validation_0-auc:1	validation_1-auc:0.908948
[44]	validation_0-auc:1	validation_1-auc:0.908499
Stopping. Best iteration:
[19]	validation_0-auc:0.999996	validation_1-auc:0.922834

print(metrics.classification_report(y_test,xgb_model.predict(X_test,ntree_limit = 19)))

             precision    recall  f1-score   support

          0       1.00      0.99      1.00     42944
          1       0.26      0.35      0.30       237

avg / total       0.99      0.99      0.99     43181

false_positive_rate, true_positive_rate, thresholds = metrics.roc_curve(y_test, xgb_model.predict(X_test,ntree_limit = 17))
roc_auc = metrics.auc(false_positive_rate, true_positive_rate)

print(metrics.roc_auc_score(y_test,xgb_model.predict(X_test,ntree_limit =19)))
0.672322791491

**Comment by mahnunchik** [comment]
Hi @nareshshah139 
I've faced with the same behaviour. Whats happen?
